schemas/fingerprint-server-api.yaml

```diff
@@ -558,8 +660,117 @@ components:
             data:
               $ref: '#/components/schemas/BotdResult'
             error:
-              $ref: '#/components/schemas/BotdError'
+              $ref: '#/components/schemas/ProductError'
+        ipInfo:
+          title: SignalResponseIpInfo
+          type: object
+          properties:
+            data:
+              $ref: '#/components/schemas/IpInfoResult'
+            error:
+              $ref: '#/components/schemas/ProductError'
+        incognito:
+          title: SignalResponseIncognito
+          type: object
+          properties:
+            data:
+              type: object
+              properties:
+                result:
+                  type: boolean
+                  description: >
+                    `true` if we detected incognito mode used in the browser, `false` otherwise.
+                  example: false
+            error:
+              $ref: '#/components/schemas/ProductError'
+        rootApps:
+          title: SignalResponseRootApps
+          type: object
+          properties:
+            data:
+              type: object
+              properties:
+                result:
+                  type: boolean
+                  description: >
+                    Android specific root management apps detection. There are 2 values:
+                    • `true` - Root Management Apps detected (e.g. Magisk)
+                    • `false` - No Root Management Apps detected
 
+                    Available only for events from Android client. The field will be empty for a browser/iOS event.
+                  example: false
+            error:
+              $ref: '#/components/schemas/ProductError'
+        emulator:
+          title: SignalResponseEmulator
+          type: object
+          properties:
+            data:
+              type: object
+              properties:
+                result:
+                  type: boolean
+                  description: >
+                    Android specific emulator detection. There are 2 values:
+                    • `true` - Emulated environment detected (e.g. launch inside of AVD)
+                    • `false` - No signs of emulated environment detected
+
+                    Available only for events from Android client. The field will be empty for a browser/iOS event.
```

Comment:
"empty" meaning `undefined` or `null` ?

Reply:
Good question, I just copied it from the documentation

Reply:
I tested it, it won't be present in the response at all. So let's say that: "The field will not be present for a browser or iOS event."

Reply:
Sure, fixed!